runtimes/eden/src/lib.rs

```diff
@@ -381,6 +384,65 @@ sp_api::impl_runtime_apis! {
 
 			impl frame_system_benchmarking::Config for Runtime {}
 
+			use xcm::latest::prelude::*;
+			use frame_benchmarking::BenchmarkError;
+			use pallet_xcm_benchmarks::generic::Pallet as XcmBenchmarks;
+			impl pallet_xcm_benchmarks::Config for Runtime {
+				type XcmConfig = xcm_config::XcmConfig;
+				type AccountIdConverter = xcm_config::LocationToAccountId;
+				fn valid_destination() -> Result<MultiLocation, BenchmarkError> {
+					Ok(MultiLocation::parent())
+				}
+				fn worst_case_holding() -> MultiAssets {
+				// 1 fungibles can be traded in the worst case
+					const HOLDING_FUNGIBLES: u32 = 1;
+					let fungibles_amount: u128 = 100;
+					let assets = (0..HOLDING_FUNGIBLES).map(|i| {
+						let location: MultiLocation = GeneralIndex(i as u128).into();
+						MultiAsset {
+							id: Concrete(location),
+							fun: Fungible(fungibles_amount * i as u128),
+						}
+						.into()
+					})
+					.chain(
+						core::iter::once(
+							MultiAsset {
+								id: Concrete(MultiLocation::parent()),
+								fun: Fungible(u128::MAX)
+							}
+						)
+					)
+					.collect::<Vec<_>>();
+					assets.into()
```

Comment:
Is this a fancy way of writing:

```
vec![
	 MultiAsset {
	      id: Concrete(GeneralIndex(0).into()),
	      fun: Fungible(0),
	},
	 MultiAsset {
	      id: Concrete(GeneralIndex(1).into()),
	      fun: Fungible(fungibles_amount),
	},
    MultiAsset { id: Concrete(MultiLocation::parent()),fun: Fungible(u128::MAX)}
 ].into()
```

Or are there some magic macros working here?

Reply:
No magic macros, the code is supposed to loop over a lot of assets but I have set it to one. You made me notice that it might be a mistake. Shall I perform the worst-case scenario with more than one asset holding register?

Reply:
`(0..HOLDING_FUNGIBLES)` will yield only one element `0`

Reply:
For fungible assets we have only one now.